sharded/shardedDB.go

```diff
@@ -40,8 +43,19 @@ func NewShardedPersister(path string, batchDelaySeconds int, maxBatchSize int, m
 	}, nil
 }
 
+func createDB(dbType storageUnit.DBType, path string, batchDelaySeconds int, maxBatchSize int, maxOpenFilesPerShard int) (types.Persister, error) {
```

Comment:
by having this function, there is no other way to reuse this component with other types other than serialDB or DB. If we will implement a new persister type, we will need to change this function also. This breaks the OCP principle.
Solution: on the NewShardedPersister you should provide a factory that can create a new persister along with the ShardIDProvider. Something like:
```GO
NewShardedPersister(persisterFactory PersisterFactory, idProvider types.ShardIDProvider)
// .... 

type PersisterFactory interface{
    Create(newPath) (types.Persister, error)
    GetBasePath() string
}
```

we can then implement this persister factory that will contain the switch inside it. The rest needed parameters can be provided on the persister factory's constructor.